lib/galaxy/auth/providers/ldap_ad.py

```diff
@@ -151,6 +156,15 @@ def ldap_search(self, email, username, options):
                 # setup search
                 attributes = [_.strip().format(**params)
                               for _ in options['search-fields'].split(',')]
+                if string_as_bool(options.get('ldap-autogenerate', False)):
+                    if str(username).lower() == 'none':
+		        attributes.append('displayName')
+                    else:
+                        attributes.append('mail')
+
+                if 'search-memberof-filter' in options:
```

Comment:
Question is if this (a specialized search filter) is actually needed or could be covered with the already present possibility of search filters.

I'm not sure of the setup you have in mind. For my case its a Galaxy that only certain LDAP users can use. They will be autoregistered and con login only if they are found using the search filters.

Reply:
In our case the user can login only if he belongs to the right group in addition to being present in our Active Directory. That allow us to manage several instances of Galaxy with a central authentication provider  and dynamically allow ours users to log into specific instances (with the same credentials) by adding or removing them from customs groups.